### ISSUE TITLE
no-extra-parens incorrectly reports async function expressions

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#9019]
**Tell us about your environment**

* **ESLint Version:** 4.3.0
* **Node Version:** 8.2.1
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
parserOptions:
  ecmaVersion: 2017
rules:
  no-extra-parens: error
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
(async function() {})

(async function() {}())
```

**What did you expect to happen?**

I expected no error to be reported, because removing the parentheses results in a syntax error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
1:1  error  Gratuitous parentheses around expression  no-extra-parens
3:1  error  Gratuitous parentheses around expression  no-extra-parens
```

**Comment by nyuszika7h** [comment]
When is this going to be published on npm? I still have this issue after `npm i --save eslint@latest`.

**Comment by not-an-aardvark** [comment]
It should already be fixed in ESLint 4.4.0 and later.
